src/v2/components/ConnectableContextMenu/index.tsx

```diff
@@ -24,6 +39,116 @@ export const ConnectableContextMenu: React.FC<Props & BoxProps> = ({
   onChangePosition,
   zIndex = 1,
   ...rest
+}) => {
+  const type = toBaseConnectableType(connectable.__typename as any)
+
+  return (
+    <ContextMenu
+      position="absolute"
+      top={8}
+      right={8}
+      zIndex={zIndex}
+      {...rest}
+    >
+      {type === 'BLOCK' && (
+        <BlockContextMenu
+          id={connectable.id.toString()}
+          connectable={connectable}
+          channel={channel}
+          onRemove={onRemove}
+          onChangePosition={onChangePosition}
+        />
+      )}
+      {type === 'CHANNEL' && (
+        <ChannelContextMenu
+          id={connectable.id.toString()}
+          connectable={connectable}
+          channel={channel}
+          onRemove={onRemove}
+          onChangePosition={onChangePosition}
+        />
+      )}
+    </ContextMenu>
+  )
+}
+
+interface BlockContextMenuContentsProps {
+  id: string
+}
+
+const BlockContextMenu: React.FC<Props & BlockContextMenuContentsProps> = ({
+  id,
+  channel,
+  onRemove,
+  onChangePosition,
+}) => {
+  const { data, loading, error } = useQuery<
+    BlokkContextMenu,
+    BlokkContextMenuVariables
+  >(blokkContextMenu, {
+    variables: {
+      id,
+    },
+  })
+
+  if (loading) return <></>
+  if (error) return <></>
+
+  const connectable = data?.blokk
+
+  return (
+    <ContextMenuContents
+      channel={channel}
+      connectable={connectable}
+      onRemove={onRemove}
+      onChangePosition={onChangePosition}
+    />
+  )
+}
+
+interface ChannelContextMenuContentsProps {
+  id: string
+}
+
+const ChannelContextMenu: React.FC<Props & ChannelContextMenuContentsProps> = ({
+  id,
+  channel,
+  onRemove,
+  onChangePosition,
+}) => {
+  const { data, loading, error } = useQuery<
+    ChannelContextMenuType,
+    ChannelContextMenuVariables
+  >(channelContextMenu, {
+    variables: {
+      id,
+    },
+  })
+
+  if (loading) return <></>
```

Comment:
Why not return a loading state here; this is going to feel really weird otherwise.